### ISSUE TITLE
Library name depends on build type capitalization

### ISSUE THREAD

**Issue by pitrou** [microsoft/mimalloc#144]
If I use:
```
cmake -DCMAKE_BUILD_TYPE=Release [...]
```
then the generated library is `libmimalloc.a` /  `libmimalloc.so` (on Linux)

But if I use:
```
cmake -DCMAKE_BUILD_TYPE=release [...]
```
then the generated library is `libmimalloc-release.a` /  `libmimalloc-release.so`.

It's a bit error-prone, especially when propagating the build type from another layer.

**Comment by devnexen** [comment]
Might be due to this https://github.com/microsoft/mimalloc/blob/afaf4924cbe02e386111e2a9d8ce209d0b0341d3/CMakeLists.txt#L93
maybe I m wrong though...

**Comment by pitrou** [comment]
Right. Personally I don't mind for the release library to be called `libmimalloc-release.a` instead of `libmimalloc.a` (it's even a bit easier for us). However, I don't think it should change depending on how you spelt the build type on the command line :-)

**Comment by daanx** [comment]
Fixed in `d7982f8` (pushing soon)
